### PR TITLE
Removed default endPoints, removed required attributes in the configu…

### PR DESCRIPTION
**Pull request by ghost**
…ration section

Extracted three methods in ETClient.cs (RetrieveBusinessUnitProperties, ValidateRequiredConfigValues, OverrideConfigValues), Removed the default values of soapEndPoint, authEndPoint and restEndPoint properties in FuelSDKConfigurationSection.cs, Removed the required=true attribute for appSignature, clientId, clientSecret, Modified the tests in FuelSDKConfigurationSectionTest due to changes in this commit

### CODE REVIEW

File: FuelSDK-CSharp/ETClient.cs [hunk 1]
```diff
@@ -42,36 +42,20 @@ public class RefreshState
             public string OrganizationId { get; set; }
             public string Stack { get; set; }
         }
+        //public ETClient()
```
Comment (sfdrogojan) [inline FuelSDK-CSharp/ETClient.cs]: Remove commented code.

File: FuelSDK-CSharp/ETClient.cs [hunk 2]
```diff
@@ -85,8 +69,16 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
             }
             else if (parameters != null && parameters.AllKeys.Contains("jwt") && !string.IsNullOrEmpty(parameters["jwt"]))
             {
+                // intru daca refreshState == null
```
Comment (sfdrogojan) [inline FuelSDK-CSharp/ETClient.cs]: Remove comment.

File: FuelSDK-CSharp/ETClient.cs [hunk 3]
```diff
@@ -105,16 +97,22 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
             }
             else
             {
+                // intru daca refreshState == null
```
Comment (sfdrogojan) [inline FuelSDK-CSharp/ETClient.cs]: Remove comment.

File: FuelSDK-CSharp/ETClient.cs [hunk 4]
```diff
@@ -148,6 +151,53 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
                 }
         }
 
+        private void ValidateRequiredConfigValues()
```
Comment (sfdrogojan) [inline FuelSDK-CSharp/ETClient.cs:144]: Please add a check for the RestEndpoint also.

**Review by sfdrogojan — CHANGES_REQUESTED** [review]

File: FuelSDK-CSharp/ETClient.cs (hunk 4 above)
Comment (ghost) [reply FuelSDK-CSharp/ETClient.cs:144]: Method refactored.

**Review by sfdrogojan — APPROVED** [review]

File: FuelSDK-CSharp/ETClient.cs [hunk 5]
```diff
@@ -86,7 +66,14 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
             else if (parameters != null && parameters.AllKeys.Contains("jwt") && !string.IsNullOrEmpty(parameters["jwt"]))
             {
                 if (string.IsNullOrEmpty(configSection.AppSignature))
```
Comment (sfdrogojan) [inline FuelSDK-CSharp/ETClient.cs:68]: Remove first if.

**Review by sfdrogojan — CHANGES_REQUESTED** [review]

File: FuelSDK-CSharp/ETClient.cs (hunk 5 above)
Comment (ghost) [reply FuelSDK-CSharp/ETClient.cs:68]: Removed the first if.

**Review by sfdrogojan — APPROVED** [review]

**Pull request merged**